src/Components/Authentication/signup.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useAuth } from '../../Context/AuthContext/authContext'
+import './authentication.css'
+
+const Signup = () => {
+  const [userDetails, setUserDetails] = useState({firstName: '',lastName: '',email: '', password: '', confirmPassword: '', terms: false})
+  const [toggleShowPassword, setToggleShowPassword] = useState(true)
+  const [toggleShowConfirmPass, setToggleShowConfirmPass] = useState(true)
+  const [error, setError] = useState({isError: false, text: ''})
+
+
+ 
+  const {signup} = useAuth();
+
+  useEffect(()=>{
+      const timeoutId = setTimeout(()=>{
+       setError({isError: false, text: ''})
+      }, 3000)
+
+      return ()=>clearTimeout(timeoutId)
+  },[error])
+
+
+  const submitHandler = (e) =>{
+      e.preventDefault()
+      const passwordValidation = /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
+
+      if(!userDetails.firstName || !userDetails.lastName || !userDetails.email || !userDetails.password || !userDetails.confirmPassword || !userDetails.terms){
+         setError({isError: true, text: 'Please Enter all the Fields!!'})
+      }
+
+      else if(!userDetails.email.includes('@')){
+          setError({isError: true, text: 'Invalid Email-Id'})
+      }
+
+      else if(!userDetails.password.match(passwordValidation)){
+          setError({isError: true, text: 'The password must be alphanumberic and atleast 5 characters long!!'})
+      }
+
+      else if(userDetails.password !== userDetails.confirmPassword){
+          setError({isError: true, text: 'Password and Confirm Password Not Match!!'})
+      }
+
+      else if(!userDetails.terms){
+          setError({isError: true, text: 'Please accept terms and conditions!!'})
+      }else{
+          signup(userDetails)
+          setUserDetails({firstName: '', lastName: '', email: '', password: '', confirmPassword: '', terms: false})
+      }
+  }
+  return (
+    <div className="wrapper">
+    {error && <p className='error-text'>{error.text}</p>}
+        <h2>Sign-Up</h2>
+        <form>
+            <div className="input-box">
+                <input type="text" placeholder="Enter your First name" name='firstName' id='name' value={userDetails.fullName} onChange={(e)=>setUserDetails({...userDetails, firstName: e.target.value})} />
```

Comment:
suggestion : you are using name attribute, so in setter you could use the [e.target.name] instead firstname, even not only but in input fields

Reply:
Yes you are right. Will do that. Thankyou for pointing it.